core/controllers/admin_test.py

```diff
@@ -85,6 +142,430 @@ def test_change_configuration_property(self):
+
+        collection_services.load_demo('0')
+        collection_rights = rights_manager.get_collection_rights('0')
+
+        self.assertFalse(collection_rights.community_owned)
+
+        with self.swap(logging, 'info', _mock_logging_function):
+            self.post_json(
+                '/adminhandler', {
+                    'action': 'reload_collection',
+                    'collection_id': '0'
+                }, csrf_token=csrf_token)
+
+        collection_rights = rights_manager.get_collection_rights('0')
+
+        self.assertTrue(collection_rights.community_owned)
+        self.assertEqual(
+            observed_log_messages,
+            [
+                '[ADMIN] %s reloaded collection 0' % self.admin_id,
+                'Collection with id 0 was loaded.'
+            ]
+        )
+
+        self.logout()
+
+    def test_handler_raises_error_with_non_int_num_dummy_exps_to_generate(self):
+        self.login(self.ADMIN_EMAIL, is_super_admin=True)
+        response = self.get_html_response('/admin')
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        with self.assertRaisesRegexp(
+            Exception, 'invalid_type is not a number'):
+            self.post_json(
+                '/adminhandler', {
+                    'action': 'generate_dummy_explorations',
+                    'num_dummy_exps_to_publish': 1,
+                    'num_dummy_exps_to_generate': 'invalid_type'
+                }, csrf_token=csrf_token)
+
+        self.logout()
+
+    def test_handler_raises_error_with_non_int_num_dummy_exps_to_publish(self):
+        self.login(self.ADMIN_EMAIL, is_super_admin=True)
+        response = self.get_html_response('/admin')
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        with self.assertRaisesRegexp(
+            Exception, 'invalid_type is not a number'):
+            self.post_json(
+                '/adminhandler', {
+                    'action': 'generate_dummy_explorations',
+                    'num_dummy_exps_to_publish': 'invalid_type',
+                    'num_dummy_exps_to_generate': 1
+                }, csrf_token=csrf_token)
+
+        self.logout()
+
+    def test_cannot_generate_dummy_explorations_in_prod_mode(self):
+        self.login(self.ADMIN_EMAIL, is_super_admin=True)
+        response = self.get_html_response('/admin')
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        prod_mode_swap = self.swap(constants, 'DEV_MODE', False)
+        assert_raises_regexp_context_manager = self.assertRaisesRegexp(
+            Exception, 'Cannot generate dummy explorations in production.')
+
+        with assert_raises_regexp_context_manager, prod_mode_swap:
+            self.post_json(
+                '/adminhandler', {
+                    'action': 'generate_dummy_explorations',
+                    'num_dummy_exps_to_generate': 10,
+                    'num_dummy_exps_to_publish': 3
+                }, csrf_token=csrf_token)
+
```

Comment:
Add an additional check here to ensure that no explorations were generated.

You can also have a test_can_generate_dummy_explorations_in_dev_mode() just above this, to show the difference.

Reply:
Actually there are a lot of tests for generating dummy exps in GenerateDummyExplorationsTest. I have moved the failure tests to GenerateDummyExplorationsTest.
> Add an additional check here to ensure that no explorations were generated.

Done